### ISSUE TITLE
Drop tables does not work in Hive connector (schemas other than default)

### ISSUE THREAD

**Issue by ameyamahajan** [prestodb/presto#15995]
Cluster is configured to have  `hive.security=sql-standard` which means Presto would depend on hive for authorization. But it fails to obey those permissions. 
A user is granted insert, delete, update permission on table but they are unable to perform DROP operations on the tables. 

> 2021-04-23T10:42:23.160-0700    ERROR   main    com.facebook.presto.server.PrestoServer Unable to create injector, see the following errors:
> 1) Configuration property 'hive.allow-drop-table' was not used
>   at com.facebook.airlift.bootstrap.Bootstrap.lambda$initialize$2(Bootstrap.java:244)
> 

Note:   `hive.allow-drop-table=true` property  cannot be used when `hive.security=sql-standard`

Related: https://github.com/prestodb/presto/issues/3912

**Comment by agrawalreetika** [comment]
Hi @ameyamahajan,
Presto enforces the authorization checks for sql-standard type security based on the roles defined in Hive Metastore.
DROP Operation would require Ownership permission on the table.
You can find more details on permission required for different Actions - https://cwiki.apache.org/confluence/display/Hive/SQL+Standard+Based+Hive+Authorization#SQLStandardBasedHiveAuthorization-Configuration

**Comment by agrawalreetika** [comment]
Here I tried it on one of my test table, Drop worked fine for me-

Table ownership details from hive-
![image](https://user-images.githubusercontent.com/7887476/116228753-54c1d980-a773-11eb-9edb-16c218c5ff9b.png)

Drop & Alter Table from Presto-
![image](https://user-images.githubusercontent.com/7887476/116229229-df0a3d80-a773-11eb-9e13-72e6701c16f5.png)

**Comment by Revellski** [comment]
@agrawalreetika when using hive.security in the default "legacy" mode there is no ownership data in Hive thus the error above.

**Comment by agrawalreetika** [comment]
@Revellski When we use the default security option i.e. 'legacy', we have to set **hive.allow-drop-table=true** configuration in hive catalog. 
Here @ameyamahajan is using **hive.security=sql-standard**

**Comment by Revellski** [comment]
@agrawalreetika correct, `hive.allow-drop-table=true` is set along with `hive.security=legacy`. The ownership information in Hive is missing, thus you end up with the same error. The workaround is to grant privileges to a user, f.e.:

```ALTER TABLE hive.<SCHEMA>.<TABLE> SET AUTHORIZATION <USER>```

If this doesn't look related I'll start a separate thread, let me know.
